x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/EsqlTypeResolutions.java

```diff
@@ -48,4 +48,8 @@ public static Expression.TypeResolution isExact(Expression e, String operationNa
     public static Expression.TypeResolution isSpatialPoint(Expression e, String operationName, TypeResolutions.ParamOrdinal paramOrd) {
         return isType(e, EsqlDataTypes::isSpatialPoint, operationName, paramOrd, "geo_point or cartesian_point");
     }
+
+    public static Expression.TypeResolution isSpatial(Expression e, String operationName, TypeResolutions.ParamOrdinal paramOrd) {
+        return isType(e, EsqlDataTypes::isSpatial, operationName, paramOrd, "geo_point or cartesian_point or geo_shape or cartesian_shape");
```

Comment:
```suggestion
        return isType(e, EsqlDataTypes::isSpatial, operationName, paramOrd, "geo_point", "cartesian_point", "geo_shape", "cartesian_shape");
```

`isType` handles automatically an array of values in a smart way.